File: app/src/main/java/yoyo/app/android/com/yoyoapp/Trip/details/TripDetailsViewModel.java
package yoyo.app.android.com.yoyoapp.Trip.details;

import android.app.Application;
import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import yoyo.app.android.com.yoyoapp.DataModels.Trip;

public class TripDetailsViewModel extends AndroidViewModel {

    private MutableLiveData<Trip> tirpMutableLiveData;
    private TripDetailsRepository tirpDetailsRepository;

    public TripDetailsViewModel(@NonNull Application application) {
        super(application);
        tirpDetailsRepository = TripDetailsRepository.getInstance(getApplication());
    }

    public void initDetails(String tripId) {
        tirpMutableLiveData = new MutableLiveData<>();
        tirpMutableLiveData = tirpDetailsRepository.getDetails(tripId);
    }

    public LiveData<Trip> getDetails() {
        return tirpMutableLiveData;
    }
}

File: app/src/main/java/yoyo/app/android/com/yoyoapp/trip/details/TripDetailsViewModel.kt
package yoyo.app.android.com.yoyoapp.trip.details

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import yoyo.app.android.com.yoyoapp.DataModels.Trip

class TripDetailsViewModel(application: Application) : AndroidViewModel(application) {

    private val tripMutableLiveData: MutableLiveData<Trip>? = null
    private val tripDetailsRepository: TripDetailsRepository

    init {
        tripDetailsRepository = TripDetailsRepository(getApplication())
    }


}
